java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/pub/IndexController.java

```diff
@@ -41,17 +62,75 @@ public FileSystemResource indexesPhp(@RequestParam(defaultValue="", required=fal
 		return new FileSystemResource(fl); 
 	}
     
-    @RequestMapping("indexes.php")
-    @ResponseBody
-    public FileSystemResource indexesPhp(@RequestParam(required=false) boolean update, 
-    		@RequestParam(required=false)  boolean refresh) throws IOException {
+    @RequestMapping(value = "indexes.php")
+    public String indexesPhp(@RequestParam(required=false) boolean update,
+    		@RequestParam(required=false)  boolean refresh, Model model) throws IOException {
     	// keep this step
     	File fl = downloadIndexes.getIndexesXml(refresh || update, false);
     	// TODO print table
     	// possible algorithm
     	// 1. read from xml into DownloadIndex Map, Map<DownloadType, List<DownloadIndex> >
-    	// 2. print each section 
+    	// 2. print each section
+		DownloadIndexDocument doc = unmarshallIndexes(fl);
+		model.addAttribute("region", doc.getMaps());
+		model.addAttribute("road_region", doc.getRoadMaps());
+		model.addAttribute("srtmcountry", doc.getSrtmMaps());
+		model.addAttribute("wiki", doc.getWikimaps());
+		model.addAttribute("wikivoyage", doc.getWikivoyages());
+        return "pub/indexes";
+    }
 
-        return new FileSystemResource(fl); 
+    @RequestMapping(value = {"list", "list.php"})
+    public String listPhp(@RequestParam(required = false) String sortby,
+                          @RequestParam(required = false) boolean asc,
+                          Model model) throws IOException {
+        File fl = null;
+        if (sortby == null || sortby.isEmpty()) {
+            // Update at first load
+            fl = downloadIndexes.getIndexesXml(true, false);
+        } else {
+            // Do not update when filtering
+            fl = downloadIndexes.getIndexesXml(false, false);
+        }
+        DownloadIndexDocument doc = unmarshallIndexes(fl);
```

Comment:
Too much copy paste, almost every line regions.stream().sorted(Comparator.comparing(DownloadIndex::getDate).... Must be a generic function for that?